.github/workflows/release.yaml

```diff
@@ -14,183 +14,207 @@ jobs:
   create-release-tag:
     runs-on: ubuntu-latest
 
+    outputs:
+      branch_name: ${{ steps.branch.outputs.branch_name }}
+
     steps:
-      - name: Setup JDK
-        uses: actions/setup-java@v1
-        with:
-          java-version: 1.8
-
-      - name: Checkout master
-        uses: actions/checkout@v2
-        with:
-          ref: master
-          token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
-
-      - name: Set up Github credentials
-        run: |
-          git config user.name Gaffer
-          git config user.email github-actions@github.com
-
-      - name: Merge develop into master
-        run: |
-          git remote set-branches --add origin develop
-          git pull
-          git merge origin/develop
-
-      - name: Set version
-        run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
-
-      - name: Update versions
-        run: |
-          mvn versions:set -DnewVersion=$RELEASE_VERSION -DgenerateBackupPoms=false
-
-      - name: Push changes
-        run: |
-          git commit -a -m "prepare release $artifactId-$RELEASE_VERSION"
-          git tag $artifactId-$RELEASE_VERSION
-          git push origin --tags
-          git push
+    - name: Setup JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    - name: Checkout master
+      uses: actions/checkout@v2
+      with:
+        ref: master
+        token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
+
+    - name: Set up Github credentials
+      run: |
+        git config user.name Gaffer
+        git config user.email github-actions@github.com
+
+    - name: Set release version
+      run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
+
+    - name: Set branch name
+      run: echo "BRANCH_NAME=$(git branch -a | grep $RELEASE_VERSION | tail -n 1 | cut -c 18-)" >> $GITHUB_ENV
+
+    - id: branch
```

Comment:
Seems to be missing a name, probably deserves one as it sets the output used in subsequent jobs and so is quite important.

Reply:
My thinking was to highlight it as an output step but I have added a name just to be consistent and clearer to all 4 release branch PRs.